#! /bin/sh
patch -p1 -f $* < $0
exit $?

2006-01-05  Werner Koch  <wk@g10code.com>

        * src/common/utils.c (w32_get_locale_dir, get_locale_dir): New.
        (get_plugin_dir) [W32]: Use it here.
        * src/common/sylpheed.c (sylpheed_init): Use it here.

--- sylpheed-claws-2.0.0/src/common/utils.c	2006-01-13 18:11:38.000000000 +0100
+++ sylpheed-claws-2.0.0/src/common/utils.c	2006-01-05 16:36:51.000000000 +0100
@@ -1979,16 +1979,51 @@
   else
     return -1;
 }
+
+/* Return a malloced string with the locale dir.  Return NULL on
+   error. */
+static char *w32_get_locale_dir(void)
+{
+        char name[MAX_PATH+35];
+        char *p;
+
+        if ( !GetModuleFileNameA (0, name, sizeof (name)-30) )
+                return NULL;
+        p = strrchr (name, '\\');
+        if (!p)
+        {
+                *name = '\\';
+                p = name;
+        }
+        p++;
+        strcpy (p, "\\share\\locale");
+        return g_strdup (name);
+}
+#endif /* G_OS_WIN32 */
+
+/* Return a static string with the locale dir. */
+const gchar *get_locale_dir(void)
+{
+	static gchar *loc_dir;
+
+#ifdef G_OS_WIN32
+	if (!loc_dir)
+                loc_dir = w32_get_locale_dir();
 #endif
+	if (!loc_dir)
+		loc_dir = LOCALEDIR;
+        
+	return loc_dir;
+}
+
 
 const gchar *get_home_dir(void)
 {
 #ifdef G_OS_WIN32
 	static char home_dir[MAX_PATH] = "";
 
-	if (home_dir[0] == '\0')
-		{
-			if (w32_shgetfolderpath
+	if (home_dir[0] == '\0') {
+                if (w32_shgetfolderpath
 			    (NULL, CSIDL_APPDATA|CSIDL_FLAG_CREATE,
 			     NULL, 0, home_dir) < 0)
 				strcpy (home_dir, "C:\\Sylpheed");
@@ -2098,7 +2133,7 @@
 	static gchar *plugin_dir = NULL;
 
 	if (!plugin_dir)
-                plugin_dir = g_strconcat(sylpheed_get_startup_dir(),
+                plugin_dir = g_strconcat(get_locale_dir(),
                                          "\\lib\\sylpheed-claws\\plugins\\",
                                          NULL);
 	return plugin_dir;
--- sylpheed-claws-2.0.0/src/common/sylpheed.c	2006-01-13 18:11:38.000000000 +0100
+++ sylpheed-claws-2.0.0/src/common/sylpheed.c	2006-01-05 16:32:59.000000000 +0100
@@ -100,7 +100,7 @@
 
 	setlocale(LC_ALL, "");
 #ifdef ENABLE_NLS
-	bindtextdomain(PACKAGE, LOCALEDIR);
+	bindtextdomain(PACKAGE, get_locale_dir () );
 	bind_textdomain_codeset (PACKAGE, "UTF-8");
 	textdomain(PACKAGE);
 #endif /*ENABLE_NLS*/
